src/main/java/edu/kis/powp/jobs2d/events/SelectHistoryListOptionListener.java

```diff
@@ -0,0 +1,33 @@
+package edu.kis.powp.jobs2d.events;
+
+import edu.kis.powp.jobs2d.command.history.HistoryEntry;
+import edu.kis.powp.jobs2d.command.manager.DriverCommandManager;
+
+import javax.swing.*;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+
+public class SelectHistoryListOptionListener implements ListSelectionListener {
+
+    private final DefaultListModel<HistoryEntry> historyCommandList;
+    private final DriverCommandManager commandManager;
+
+    public SelectHistoryListOptionListener(DriverCommandManager commandManager, DefaultListModel<HistoryEntry> historyCommandList) {
+        this.commandManager = commandManager;
+        this.historyCommandList = historyCommandList;
+    }
+
+    @Override
+    public void valueChanged(ListSelectionEvent e) {
+        if (e.getValueIsAdjusting() == false) {
+            final int selectedIndex = ((JList<HistoryEntry>) e.getSource()).getSelectedIndex();
```

Comment:
Casting JList is a worse variant. My tip concerning DefaultListModel and ListSelectionEvent was probably incorrect.
Nevertheless, it is possible to avoid the direct dependency with JList, by introducing a new abstraction (an interface) which will encapsulate retrieving a selected index.